packages/core/vue/src/useWallet.ts

```diff
@@ -0,0 +1,225 @@
+            window.open(wallet.value.url, '_blank');
+            throw newError(new WalletNotReadyError());
+        }
+
+        try {
+            connecting.value = true;
+            await adapter.value.connect();
+        } catch (error: any) {
+            walletName.value = null;
+            throw error;
+        } finally {
+            connecting.value = false;
+        }
+    };
+
+    // Disconnect the adapter from the wallet.
+    const disconnect = async (): Promise<void> => {
+        if (disconnecting.value) return;
+        if (! adapter.value) {
+            walletName.value = null;
+            return;
+        }
+
+        try {
+            disconnecting.value = true;
+            await adapter.value.disconnect();
+        } finally {
+            walletName.value = null;
+            disconnecting.value = false;
+        }
+    };
+
+    // Send a transaction using the provided connection.
+    const sendTransaction = async (transaction: Transaction, connection: Connection, options?: SendTransactionOptions) => {
+        if (! adapter.value) throw newError(new WalletNotSelectedError());
+        if (! connected.value) throw newError(new WalletNotConnectedError());
+        return await adapter.value.sendTransaction(transaction, connection, options);
+    };
+
+    // Sign a transaction if the wallet supports it.
+    const signTransaction = async (transaction: Transaction) => {
+        if (! adapter.value) throw newError(new WalletNotSelectedError());
+        if (! connected.value) throw newError(new WalletNotConnectedError());
+        if (!('signTransaction' in adapter.value)) throw newError(new OperationNotSupportedByWalletError());
+        return await adapter.value.signTransaction(transaction);
+    };
+
+    // Sign multiple transactions if the wallet supports it
+    const signAllTransactions = async (transactions: Transaction[]) => {
+        if (! adapter.value) throw newError(new WalletNotSelectedError());
+        if (! connected.value) throw newError(new WalletNotConnectedError());
+        if (!('signAllTransactions' in adapter.value)) throw newError(new OperationNotSupportedByWalletError());
+        return await adapter.value.signAllTransactions(transactions);
+    };
+
+    // Sign an arbitrary message if the wallet supports it.
+    const signMessage = async (message: Uint8Array) => {
+        if (! adapter.value) throw newError(new WalletNotSelectedError());
+        if (! connected.value) throw newError(new WalletNotConnectedError());
+        if (!('signMessage' in adapter.value)) throw newError(new OperationNotSupportedByWalletError());
+        return await adapter.value.signMessage(message);
+    };
+
+    // If autoConnect is enabled, try to connect when the adapter changes and is ready.
+    watchEffect(async (): Promise<void> => {
+        if (! autoConnect || ! adapter.value || ! ready.value || connected.value || connecting.value) return;
+        await connect();
+    });
```

Comment:
The implementation has changed significantly in the React package:
https://github.com/solana-labs/wallet-adapter/blob/581cb52a41ad7a255542de39115d051b0833e57c/packages/core/react/src/WalletProvider.tsx#L70-L87

Several types of bugs were eliminated. This also shouldn't call the `connect` method because that can open a popup if the wallet isn't available.